### PR TITLE
Rh starting on UI

### PR DESCRIPTION
**Pull request by reherr**

### CODE REVIEW

**Review by borvux — APPROVED** [review]
I think it looks good

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```
Comment (JoseMDO) [inline app/controllers/application_controller.rb:2]: Nice!

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -6,15 +6,28 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
+    <%= render partial: "shared/cdn_assets" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:9]: Great job adding your cdn assets into a partial! This helps reduce the amount of code in the application layout file :)

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -6,15 +6,28 @@
...
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_importmap_tags %>
   </head>
 
   <body>
 
-  <p class="notice"><%= notice %></p>
-       <p class="alert"><%= alert %></p>
-       
+    <%= render partial: "shared/navbar" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:17]: Same here :)

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -6,15 +6,28 @@
...
+
+  <div class="container">
+      <% if notice.present? %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:20]: Great job adding your notices here to reduce adding them inside of your code!

File: app/views/shared/_navbar.html.erb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <ul class="navbar-nav mr-auto">
+        <li class="nav-item dropdown">
+          <a class="nav-link dropdown-toggle" href="#" role="button" data-bs-toggle="dropdown" aria-expanded="false">
+            Resources
+          </a>
+          <div class="dropdown-menu">
+            <a class="dropdown-item" href="/photos">Photos</a>
```
Comment (JoseMDO) [inline app/views/shared/_navbar.html.erb:15]: Just one tip, remember now you can start using the ruby helper methods like link_to to help with not having to change the link if it ever changes

File: lib/tasks/dev.rake [hunk 6]
```diff
@@ -10,6 +10,20 @@ task({ :sample_data => :environment }) do
...
+  usernames << "bob"
+
+  usernames.each do |username|
+    User.create(
+      email: "#{username}@example.com",
+      password: "password",
+      username: username.downcase,
+      private: [true, false].sample,
+    )
+  end
```
Comment (JoseMDO) [inline lib/tasks/dev.rake:25]: Great job on the sample data task!

**Review by JoseMDO — APPROVED** [review]
Great job on your PR Rebecca! Everything looks great :)

**Pull request merged**